lib/less/rails/railtie.rb

```diff
@@ -14,11 +14,16 @@ class Railtie < ::Rails::Railtie
         require 'less-rails'
         Sprockets::Engines #force autoloading
 
-        if Sprockets.respond_to?(:register_engine)
-          args = ['.less', LessTemplate]
-          args << { mime_type: 'text/less', silence_deprecation: true } if Sprockets::VERSION.start_with?("3")
-          env.register_engine(*args)
-        end
+        config.assets.configure do |env|
+          if env.respond_to?(:register_engine)
+            args = ['.less', LessTemplate]
+            args << { mime_type: 'text/less', silence_deprecation: true } if env::VERSION.start_with?("3")
+            env.register_engine(*args)
+          elsif env.respond_to?(:register_transformer)
+            env.register_mime_type 'text/less', extensions: ['.less'], charset: :css
```

Comment:
Could you please explain why `charset` is set to `:css`? [Here](https://github.com/rails/sprockets/search?utf8=%E2%9C%93&q=register_mime_type) is examples of using `register_mime_type` method.